apps/governance/src/routes/rewards/epoch-total-rewards/epoch-total-rewards.tsx

```diff
@@ -1,21 +1,60 @@
-import { AsyncRenderer } from '@vegaprotocol/ui-toolkit';
+import { useState, useCallback, useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { AsyncRenderer, Pagination } from '@vegaprotocol/ui-toolkit';
+import type { EpochFieldsFragment } from '../home/__generated__/Rewards';
 import { useEpochAssetsRewardsQuery } from '../home/__generated__/Rewards';
-import { useRefreshAfterEpoch } from '../../../hooks/use-refresh-after-epoch';
 import { generateEpochTotalRewardsList } from './generate-epoch-total-rewards-list';
-import { NoRewards } from '../no-rewards';
 import { EpochTotalRewardsTable } from './epoch-total-rewards-table';
 
-export const EpochTotalRewards = () => {
+const EPOCHS_PAGE_SIZE = 10;
+
+type EpochTotalRewardsProps = {
+  currentEpoch: EpochFieldsFragment;
+};
+
+export const EpochTotalRewards = ({ currentEpoch }: EpochTotalRewardsProps) => {
+  // we start from the previous epoch when displaying rewards data, because the current one has no calculated data while ongoing
+  const epochId = Number(currentEpoch.id) - 1;
+  const totalPages = Math.ceil(epochId / EPOCHS_PAGE_SIZE);
+  const { t } = useTranslation();
+  const [page, setPage] = useState(1);
   const { data, loading, error, refetch } = useEpochAssetsRewardsQuery({
+    notifyOnNetworkStatusChange: true,
     variables: {
-      epochRewardSummariesPagination: {
-        first: 10,
+      epochRewardSummariesFilter: {
+        fromEpoch: epochId - EPOCHS_PAGE_SIZE,
       },
     },
   });
-  useRefreshAfterEpoch(data?.epoch.timestamps.expiry, refetch);
 
-  const epochTotalRewardSummaries = generateEpochTotalRewardsList(data) || [];
+  const refetchData = useCallback(
+    async (toPage?: number) => {
+      const targetPage = toPage ?? page;
+      await refetch({
+        epochRewardSummariesFilter: {
+          fromEpoch: Math.max(0, epochId - EPOCHS_PAGE_SIZE * targetPage),
+          toEpoch: epochId - EPOCHS_PAGE_SIZE * targetPage + EPOCHS_PAGE_SIZE,
```

Comment:
Suggestion (non-blocking): I think this would be worth annotating as the maths here is a bit opaque

Reply:
i've put this into a separate file and added a line above trying to explain what's going on. thanks!